src/khoj/processor/conversation/utils.py

```diff
@@ -110,7 +110,7 @@ def generate_chatml_messages_with_context(
         logger.debug(
```

Comment:
I forgot to actually truncate the last message if the last message exceeds max prompt size! 🙈
It'd be great if you can add that here as well. 

The logic will be a little more complicated than just `encoder.decode(encoder.encode(last_message.content)[:max_prompt_size[model_name]]` as the last bit of each message contains the actual question. Sharing sample diff for reference here:

```diff
-        last_message = messages[-1]
-        truncated_message = encoder.decode(encoder.encode(last_message.content))
+        last_message = '\n'.join(messages[-1].content.split("\n")[:-1])
+        original_question = '\n'.join(messages[-1].content.split("\n")[-1:])
+        truncated_message = encoder.decode(encoder.encode(last_message)[:max_prompt_size[model_name]]).strip()
         logger.debug(
             f"Truncate last message to fit within max prompt size of {max_prompt_size[model_name]} supported by {model_name} model:\n {truncated_message}"
         )
-        messages = [ChatMessage(content=[truncated_message], role=last_message.role)]
+        messages = [ChatMessage(content=truncated_message + original_question, role=messages[-1].role)]
```